### ISSUE TITLE
Deployment experience improvements

### ISSUE THREAD

**Issue by fiveisprime** [microsoft/vscode-azurestorage#344]
- [ ] 	Automatically configure predeploy task (or clearly document this)
- [ ] Add support for deploy subpath
- [ ] Improve feedback when creating a storage account (long running notifications for creation, I'll spec this out)
- [ ] Add support for `defaultWebAppToDeploy` similar to how App Service does this today
- [x] Basic creation mode
    - [x] Create resource group in West US
    - [x] Create set index.html for index and error documents when user chooses to enable static hosting

**Comment by burkeholland** [comment]
Suggestion: Add commands (both to the palette and right-click) for "Create Static Website".

Right now the user has to know to pick "Configure Static Website", which seems like a weird choice if you don't already have one. Otherwise, they have to right-click and create a new Storage Account, know to select Blob and then know to do the Configure step.

>   * [ ]  Create set index.html for index and error documents when user chooses to enable static hosting

What about changing the placeholder text to "Set to index.html if you have a single page application".

**Comment by PrashanthCorp** [comment]
> - [ ] 	Automatically configure predeploy task (or clearly document this)
We could leverage the task discovery that vscode uses in `Tasks: Config default build task`.

**Comment by johnpapa** [comment]
Here are some thoughts on the reduction of prompts, based on my chat with Matt recently

The experience with Azure Storage was this:

- I started with my account, so login was already there. 
- in vscode extension I right clicked my dist folder, chose "push to storage"
- was prompted for a storage account (created new one)
- prompted for a resource group (created new one)
- was asked if I wanted to enable web site (chose yes)
- was prompted for default page (typed index.html )
- was prompted for error page (typed 404.html )
- was prompted to browse to the site (took about 15 seconds before I got this prompt)

https://papapeacockstorage.z13.web.core.windows.net/

Overall fast. But some prompts feel extraneous (index.html could be assumed with fallbacks of the other common ones, IMO)

Total time was about 2 minutes.

pros - the site is fast, it didnt take long to get it running at all

cons - seems like we could default even more of these prompts for storage, resource group, default page, error page. I also already had a login ... so I imagine the time to get this working is much longer if I had to sign up from scratch.

Super nice to have ... I think it would be great if the name of the url is cleaner and more fun with such as dancing-penguins-xxxx.azurewebsites.net
